src/bigquery/src/iterator.rs

```diff
@@ -0,0 +1,110 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use bigquery_v2::client::JobService;
+use bigquery_v2::model::DataFormatOptions;
+use std::collections::VecDeque;
+use std::sync::Arc;
+
+use crate::Result;
+use crate::query::Query;
+use crate::row::Row;
+use crate::schema::Schema;
+
+pub struct RowIterator {
```

Comment:
There's a few other properties that we've found useful to expose:

All the ones from https://cloud.google.com/bigquery/docs/reference/rest/v2/jobs/query#response-body / https://cloud.google.com/bigquery/docs/reference/rest/v2/jobs/getQueryResults#response-body are the main ones. totalSlotMs from jobs.query was an important recent customer ask, for example.

Also some from the Job resource, like destination table. I suppose we could make the user call jobs.get to fetch that, though, to keep this interface smaller.

Reply:
yeah, we are leaning on letting the use access the job information and query for more if needed with `jobs.get`, but good point on that, need to document that better.